nx/test/nx/defn/grad_test.exs

```diff
@@ -1746,6 +1748,80 @@ defmodule Nx.Defn.GradTest do
     end
   end
 
+  describe "put_slice" do
+    defn grad_mean_put_slice_operand(t1, t2), do: grad(t1, &Nx.mean(Nx.put_slice(&1, t2, [0, 1])))
```

Comment:
We need similar grad tests for slice:

1. when the grad is the index
2. with dynamic indexes and strides